src/supervised/distance.jl

```diff
@@ -108,6 +108,40 @@ islipschitzcont_deriv(::L2DistLoss) = true
 isconvex(::L2DistLoss) = true
 isstronglyconvex(::L2DistLoss) = true
 
+
+# ===========================================================
+# L(y, t) = 1 - cos((y-t)*2π/c)
+
+"""
+`PeriodicLoss <: DistanceLoss`
+
+Measures distance on a circle of specified circumference.
+""" 
+immutable PeriodicLoss{T<:AbstractFloat} <: DistanceLoss
+    k::T   # k = 2π/circumference
+    PeriodicLoss(circ::T) = new(T(2π/circ))
```

Comment:
Does the loss make sense with negative circ? It doesn't look like it would cause numerical issues. But I would at least check for `0` like here: https://github.com/JuliaML/Losses.jl/blob/master/src/supervised/distance.jl#L118 ?